src/test/java/jnr/posix/FileStatTest.java

```diff
@@ -157,4 +155,12 @@ public void filestatFailed() throws Throwable {
             }
         }
     }
+
+    @Test public void statIsWritable() throws Throwable {
+      String tempDirectory = System.getenv("TEMP");
+      FileStat st = posix.stat(tempDirectory);
+
+      // FIXME: This fails on JRuby, but succeeds here. Why?
```

Comment:
Are you sure win8 and win10 have different values?  I am not sure if this platform work is needed or not?

I did not come up with a fix but I thought I confirmed that CommonFileInformation.getMode() was returning the wrong number.  Seems like perhaps something with this modulo logic is wrong or something.  If you look at MRI source code it feels like I followed that logic but something is not right. 

To recap, I think this is not platform specific at all and I think the mode calculated is not right.  The mode is only done in the getMode() method mentioned above and also in WindowsRawFileStat.setup.  Looking at how MRI calculates this will be helpful because I looked to them for guidance on getting these unix mode values right...